pkg/reconciler/sequence/sequence.go

```diff
@@ -121,59 +119,59 @@ func (r *Reconciler) reconcile(ctx context.Context, p *v1alpha1.Sequence) error
 	//    than channel, we could just (optionally) feed it directly to the following subscription.
 	// 3. Rinse and repeat step #2 above for each Step in the list
 	// 4. If there's a Reply, then the last Subscription will be configured to send the reply to that.
-	if p.DeletionTimestamp != nil {
+	if s.DeletionTimestamp != nil {
 		// Everything is cleaned up by the garbage collector.
 		return nil
 	}
 
-	channelResourceInterface := r.DynamicClientSet.Resource(duckroot.KindToResource(p.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())).Namespace(p.Namespace)
-
+	gvr, _ := meta.UnsafeGuessKindToResource(s.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())
+	channelResourceInterface := r.DynamicClientSet.Resource(gvr).Namespace(s.Namespace)
 	if channelResourceInterface == nil {
-		msg := fmt.Sprintf("Unable to create dynamic client for: %+v", p.Spec.ChannelTemplate)
+		msg := fmt.Sprintf("Unable to create dynamic client for: %+v", s.Spec.ChannelTemplate)
 		logging.FromContext(ctx).Error(msg)
 		return errors.New(msg)
 	}
 
 	// Tell tracker to reconcile this Sequence whenever my channels change.
-	track := r.resourceTracker.TrackInNamespace(p)
+	track := r.channelableTracker.TrackInNamespace(s)
 
-	channels := make([]*duckv1alpha1.Channelable, 0, len(p.Spec.Steps))
-	for i := 0; i < len(p.Spec.Steps); i++ {
-		ingressChannelName := resources.SequenceChannelName(p.Name, i)
-		c, err := r.reconcileChannel(ctx, ingressChannelName, channelResourceInterface, p)
-		if err != nil {
-			logging.FromContext(ctx).Error(fmt.Sprintf("Failed to reconcile Channel Object: %s/%s", p.Namespace, ingressChannelName), zap.Error(err))
-			return err
+	channels := make([]*duckv1alpha1.Channelable, 0, len(s.Spec.Steps))
+	for i := 0; i < len(s.Spec.Steps); i++ {
+		ingressChannelName := resources.SequenceChannelName(s.Name, i)
 
+		channelObjRef := corev1.ObjectReference{
+			Kind:       s.Spec.ChannelTemplate.Kind,
+			APIVersion: s.Spec.ChannelTemplate.APIVersion,
+			Name:       ingressChannelName,
+			Namespace:  s.Namespace,
+		}
+		// Track channels and enqueue sequence when they change.
+		if err := track(channelObjRef); err != nil {
+			logging.FromContext(ctx).Error("Unable to track changes to Channel", zap.Error(err))
```

Comment:
```suggestion
			logging.FromContext(ctx).Error("Unable to track changes to Channel", zap.Error(err), zap.Any("channel", channelObjRef))
```

Reply:
annotated the error as before and returned, thanks!